lib/api/src/rest/schema.rs

```diff
@@ -651,3 +653,70 @@ pub struct QueryGroupsRequest {
 
     pub shard_key: Option<ShardKeySelector>,
 }
+
+#[derive(Serialize, Deserialize, JsonSchema, Validate, Debug, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub struct SearchMatrixRequestInternal {
+    /// Look only for points which satisfies this conditions
+    #[validate]
+    pub filter: Option<Filter>,
+    /// How many points to select and search within.
+    #[validate(range(min = 1))]
```

Comment:
I assume a sane minimum sample size would be 2, since 1 won't have another vector to compare against